src/main/java/ari/parser/Parser.java

```diff
@@ -0,0 +1,295 @@
+import ari.command.Command;
+import ari.command.DeadlineCommand;
+import ari.command.DeleteCommand;
+import ari.command.EventCommand;
+import ari.command.FindCommand;
+import ari.command.IncorrectCommand;
+import ari.command.ListCommand;
+import ari.command.MarkCommand;
+import ari.command.TodoCommand;
+import ari.command.UnmarkCommand;
+import ari.exception.CommandFormatException;
+import ari.exception.EmptyCommandException;
+
+/**
+ * Deals with making sense of the USER command/input
+ */
+public class Parser {
+    public Parser() {
+    }
+
+    /**
+     * Assigns method to execute based on user input
+     *
+     * @param command user input: command and description
+     * @return command to execute
+     */
+    public Command parse(String command) {
+        String[] cmdArray = command.split(" ");
+
+        String commandWord = cmdArray[0].toLowerCase();
+
+        switch (commandWord) {
+        case ByeCommand.COMMAND_WORD:
+            return new ByeCommand();
+        case ListCommand.COMMAND_WORD:
+            return new ListCommand();
+        case MarkCommand.COMMAND_WORD:
+            return prepareMark(command);
+        case UnmarkCommand.COMMAND_WORD:
+            return prepareUnmark(command);
+        case TodoCommand.COMMAND_WORD:
+            return prepareTodo(command);
+        case DeadlineCommand.COMMAND_WORD:
+            return prepareDeadline(command);
+        case EventCommand.COMMAND_WORD:
+            return prepareEvent(command);
+        case DeleteCommand.COMMAND_WORD:
+            return prepareDelete(command);
+        case FindCommand.COMMAND_WORD:
+            return prepareFind(command);
+        default:
+            return new IncorrectCommand("I am sorry Master, I am afraid I do not know what you mean");
+        }
+    }
+
+    /**
+     * Assigns methods to execute from input of save file if it exists
+     *
+     * @param instruction command to run
+     * @param description description of task
+     * @return command to execute
+     */
+    public Command fileParse(String instruction, String description) {
+        String command = instruction + " " + description;
+
+        switch (instruction) {
+        case MarkCommand.COMMAND_WORD:
+            return prepareMark(command);
+        case TodoCommand.COMMAND_WORD:
+            return prepareTodo(command);
+        case DeadlineCommand.COMMAND_WORD:
+            return prepareDeadline(command);
+        case EventCommand.COMMAND_WORD:
+            return prepareEvent(command);
+        default:
+            return new IncorrectCommand("Unknown commands in file!");
+        }
+    }
+
+    /**
+     * Splits file command
+     *
+     * @param fileCommand input from file
+     * @return an array containing the command, whether to mark the task and task description
+     */
+    public String[] fileBreakdown(String fileCommand) {
+        String[] words = fileCommand.split(" ");
+        String[] returnCommands = new String[3];
```

Comment:
```suggestion
        String[] commandArguments= new String[3];
```
Is the variable name returnCommands unintuitive? Maybe something like commandArugments would make its purpose clearer